tools/test-harness/postgres_gis_ha_test.go

```diff
@@ -0,0 +1,172 @@
+package tests
+
+import (
+	"testing"
+	"encoding/json"
+)
+
+// Test the Postgres HA container by running the relevant example and validated the
+// various resources are created and running with the correct statuses
+func TestPostgresGISHA(t *testing.T) {
+	t.Parallel()
+	t.Log("Testing the 'postgres-gis-ha' example...")
+	harness := setup(t, timeout, true)
+	var pods []string
+	env := []string{}
+
+	t.Run("Run the Postgres HA container example", func(t *testing.T) {
+		_, err := harness.runExample("examples/kube/postgres-gis-ha/run.sh", env, t)
+		if err != nil {
+			t.Fatalf("Could not run example: %s", err)
+		}
+	})
+
+	// Validate that deployments are created, that relevant pods are created for those deployments
+	// and validates that they are ready to use
+	t.Run("Checking pods statuses", func(t *testing.T) {
+		t.Log("Checking if postgres-gis-ha-01 deployment is ready...")
+		if ok, err := harness.Client.IsDeploymentReady(harness.Namespace, "postgres-gis-ha-01"); !ok {
+			t.Fatal(err)
+		}
+
+		t.Log("Checking if postgres-gis-ha-02 deployment is ready...")
+		if ok, err := harness.Client.IsDeploymentReady(harness.Namespace, "postgres-gis-ha-02"); !ok {
+			t.Fatal(err)
+		}
+
+		ha01, err := harness.Client.GetDeploymentPods(harness.Namespace, "postgres-gis-ha-01")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		ha02, err := harness.Client.GetDeploymentPods(harness.Namespace, "postgres-gis-ha-02")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if len(ha01) == 0 {
+			t.Fatal("Postgres HA #1 deployment ready but no pods found")
+		}
+
+		if len(ha02) == 0 {
+			t.Fatal("Postgres HA #2 deployment ready but no pods found")
+		}
+
+		for _, pod := range ha01 {
+			pods = append(pods, pod)
+		}
+		for _, pod := range ha02 {
+			pods = append(pods, pod)
+		}
+
+		t.Log("Checking if Postgres HA pods are ready to use")
+		if err := harness.Client.CheckPods(harness.Namespace, pods); err != nil {
+			t.Fatal(err)
+		}
+	})
+
+
+	t.Run("Testing database connectivity", func(t *testing.T) {
+		local, remote := randomPort(), 5432
+		proxy, err := harness.setupProxy(pods[0], local, remote)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer proxy.Close()
+
+		db, err := harness.setupDB("userdb", proxy.Hostname, "postgres", "password", "disable", local)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Check that extensions are installed
+		extensions, err := db.AllExtensions()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if len(extensions) < 1 {
+			t.Fatalf("extensions less then 1, it shouldn't be: %d", len(extensions))
+		} else {
+			t.Logf("Number of extensions: %v",len(extensions))
+		}
+
+		// Search through the installed extensions. If postgis is not installed
+		// the test fails
+		foundGIS := false
+		for _, extension := range extensions {
```

Comment:
This has bitten us before where this passes but we don't exercise the extension to see if it actually works. Should we be doing something a bit more "real" here

Reply:
@thesteve0 We could do a bit more. In one of the tests we have, we call `SELECT postgis_full_version();` and that could be applied here as well.

Please feel free to submit a patch :wink: